src/rules/graphql/no-more-than-100-fields.ts

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { Kind, FieldNode } from 'graphql';
+import { GraphQLESLintRule, GraphQLESLintRuleContext } from '@graphql-eslint/eslint-plugin';
+import { getClosestAncestorByType, getEntityNodeForEdges } from '../../util/graphql-ast-utils';
+
+export const NO_MORE_THAN_100_FIELDS_RULE_ID = 'offline-graphql-no-more-than-100-fields';
+
+export const rule: GraphQLESLintRule = {
+    meta: {
+        type: 'suggestion',
+        hasSuggestions: false,
+        docs: {
+            category: 'Operations',
+            description:
+                'For the entity with more than 100 fields, server might not return all the expected records',
```

Comment:
Is it 'might'? Or 100 a known hard threshold?

Reply:
100 is a hard threshold. 
See https://salesforce.quip.com/F6zRAZq4UqVD